tests/python/unittest/test_numpy_ndarray.py

```diff
@@ -1401,3 +1401,23 @@ def test_from_numpy_exception():
     np_array = _np.array([[1, 2], [3, 4], [5, 6]], dtype="float32")
     mx_array = mx.npx.from_numpy(np_array, zero_copy=False)
     np_array[2, 1] = 0 # no error
+
+def test_mixed_array_types():
+    np_array = _np.array([[1, 2], [3, 4], [5, 6]], dtype="float32")
+    mx_array = mx.np.ones((3, 1))
+    assert_almost_equal(mx_array + np_array, 1+np_array)
+
+def test_mixed_array_types_share_memory():
+    np_array = _np.array([[1, 2], [3, 4], [5, 6]], dtype="float32")
+    mx_array = mx.npx.from_numpy(np_array)
+    assert _np.may_share_memory(np_array, mx_array)
```

Comment:
i have a stupid numpy question - usually in what case would `may_share_memory` be useful?

Reply:
Something like this:
https://github.com/scikit-learn/scikit-learn/blob/36ae6a06d690b992c21280708ca365fce5beec99/sklearn/utils/validation.py#L662-L663